parsers/ESIOS.py

```diff
@@ -90,9 +88,13 @@ def fetch_exchange(
 
         net_flow *= EXCHANGE_MULTIPLICATION_FACTOR_MAP[zone_key]
 
+        exchange_datetime = datetime.strptime(
+            value["datetime_utc"], "%Y-%m-%dT%H:%M:%SZ"
+        ).replace(tzinfo=TIMEZONE)
```

Comment:
The timezone here should be UTC as the value sent from the server is in Zulu (Z).

I would also look into using `datetime.fromisofomat()` instead. But since we don't we are still on Python 3.10 it can't parse zulu directly and you'll have to replace the Z with `+00:00` which means the same thing but uses the extended ISO format.

Reply:
so okay for the current python verion in use i need to manually replace the z with +00:00
![fix](https://github.com/user-attachments/assets/6c60fdad-b901-494d-bee9-691c8b875a50)

this seems better ?

Reply:
Yeah it's simpler, then we can just remove the replacement when we upgrade our python version.